### ISSUE TITLE
separate docker run --rm tasks kill each other

### ISSUE THREAD

**Issue by mcandre** [docker/docker#13341]
I want to run multiple, unrelated tasks using `docker --rm ...`, but when I start one, it kills the other.

Steps to reproduce:

Terminal A

```
docker run --rm --privileged -v $(pwd):/mnt -t mcandre/docker-debian:woody sh -c 'apt-get update && apt-get install -y debootstrap && mkdir /chroot && debootstrap --arch i386 potato /chroot http://archive.debian.org/debian && cd /chroot && tar czvf /mnt/rootfs.tar.gz .'
```

Terminal B

```
$ docker run --rm --privileged -v $(pwd):/mnt -t mcandre/docker-ubuntu:hardy sh -c 'apt-get update && apt-get install -y debootstrap && mkdir /chroot && debootstrap edgy /chroot http://old-releases.ubuntu.com/ubuntu/ && cd /chroot && tar czvf /mnt/rootfs.tar.gz .'
...
```

Expected:

The docker commands in A and B continue until their respective processes complete.

Observed:

The terminal A docker command dies with:

```
FATA[0032] Error response from daemon: Cannot destroy container fec18cd258843997f41cec363cd2d9ad87516d4770baede34358829021dbc789: Failed to set container state to RemovalInProgress: Status is already RemovalInProgress
```

System:

```
$ specs docker os
specs --version
0.19

docker version
Client version: 1.6.0
Client API version: 1.18
Go version (client): go1.4.2
Git commit (client): 4749651
OS/Arch (client): darwin/amd64
Server version: 1.6.0
Server API version: 1.18
Go version (server): go1.4.2
Git commit (server): 4749651
OS/Arch (server): linux/amd64

boot2docker version
Boot2Docker-cli version: v1.6.0
Git commit: 9894ae9

vagrant --version
Vagrant 1.7.2

vboxwebsrv --help 2>&1 | grep VirtualBox
Oracle VM VirtualBox web service Version 4.3.24

system_profiler SPSoftwareDataType | grep 'System Version'
    System Software Overview:
      System Version: OS X 10.10.3 (14D136)
```

**Comment by mcandre** [comment]
Here's the rest of the information:

```
$ docker version
Client version: 1.6.0
Client API version: 1.18
Go version (client): go1.4.2
Git commit (client): 4749651
OS/Arch (client): darwin/amd64
Server version: 1.6.0
Server API version: 1.18
Go version (server): go1.4.2
Git commit (server): 4749651
OS/Arch (server): linux/amd64

$ docker info
Containers: 1
Images: 182
Storage Driver: aufs
 Root Dir: /mnt/sda1/var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 184
 Dirperm1 Supported: true
Execution Driver: native-0.2
Kernel Version: 3.18.11-tinycore64
Operating System: Boot2Docker 1.6.0 (TCL 5.4); master : a270c71 - Thu Apr 16 19:50:36 UTC 2015
CPUs: 4
Total Memory: 1.961 GiB
Name: boot2docker
ID: MCKR:WP3W:YXR7:M66R:E4QC:CSQZ:ZQAJ:E6D7:TXNH:P7J5:2RAT:TB4K
Debug mode (server): true
Debug mode (client): false
Fds: 19
Goroutines: 32
System Time: Wed May 20 22:35:38 UTC 2015
EventsListeners: 0
Init SHA1: 9145575052383dbf64cede3bac278606472e027c
Init Path: /usr/local/bin/docker
Docker Root Dir: /mnt/sda1/var/lib/docker
Username: mcandre
Registry: [https://index.docker.io/v1/]

$ uname -a
Darwin mockingjay.local 14.3.0 Darwin Kernel Version 14.3.0: Mon Mar 23 11:59:05 PDT 2015; root:xnu-2782.20.48~5/RELEASE_X86_64 x86_64
```

**Comment by ewindisch** [comment]
Cannot reproduce with 1.6.1

**Comment by cpuguy83** [comment]
I can't repro either.
Also doesn't make sense being that `--rm` is a client-side feature.

If you can provide a simple repro casting here.
Closing this now.

**Comment by adamtheturtle** [comment]
It seems like https://github.com/docker/docker/commit/c4e49d10143178c718178ff7ce857f4f8ed46a0b will fix this in Docker 1.9.0.
